### PR TITLE
adding test infrastructure and two tests

### PR DESCRIPTION
**Pull request by ChronosSF**

### CODE REVIEW

File: test/transformation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { expect } from "chai";
+import { SearchHandler, defaultFileGroups, allLangs, allPlatforms } from "./testing-library";
+
+describe("Transformation Tests", () => {
+    it("should not have any files with leftover reserved words", async () => {
+        const result = await SearchHandler.searchInFiles(
+            [/\{Platform\}|\{ProductName\}/gm],
```
Comment (HUSSAR-mtrela) [inline test/transformation.spec.ts:7]: you should get a list of variables from [docConfig.json](https://github.com/IgniteUI/igniteui-xplat-docs/blob/vnext/docConfig.json#L6) where we define them in 4 places (under each platform)

File: test/transformation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            allPlatforms,
+            allLangs,
+            defaultFileGroups.allMarkdown);
+        expect(result).to.eql([]);
+    });
+
+    it("should not have any files with leftover component replacement words", async () => {
+        const result = await SearchHandler.searchInFiles(
+            [/ComponentSelector|ComponentName/gm],
```
Comment (HUSSAR-mtrela) [inline test/transformation.spec.ts:16]: add "ComponentTitle"

File: test/testing-library.ts [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+            const res = path.resolve(dir, entity.name);
+            if (entity.isDirectory()) {
+                yield* this.getFiles(res);
+            } else {
+                yield res;
+            }
+        }
+    }
+
+    private static shouldTestFile(file: string, fileWildcards?: string[]): boolean {
```
Comment (HUSSAR-mtrela) [inline test/testing-library.ts:93]: typo "shouldTestFail" instead of "shouldTestFile"

File: test/testing-library.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+    "tocOnly": string[]
+}
+
+export type Platform = ('Blazor'|'WC'|'React')
+export type Language = ('jp'|'kr'|'en')
+
+export const productPathMap: Map<Platform, string> = new Map<Platform, string>([
+    ['React', '../dist/React/{lang}/components'],
+    ['WC', '../dist/WebComponents/{lang}/components'],
+    ['Blazor', '../dist/Blazor/{lang}/components'],
```
Comment (HUSSAR-mtrela) [inline test/testing-library.ts:17]: add: ['Angular', '../dist/Angular/{lang}/components'],

File: test/testing-library.ts [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+]);
+
+export const allPlatforms: Platform[] = ['Blazor', 'React', 'WC'];
```
Comment (HUSSAR-mtrela) [inline test/testing-library.ts:20]: add 'Angular'

**Review by HUSSAR-mtrela — CHANGES_REQUESTED** [review]

File: test/testing-library.ts (hunk 3 above)
Comment (ChronosSF) [reply test/testing-library.ts:93]: Nah, this returns if the file should be tested based on the wildcard filter you can send to the searchInFiles. I'll add some additional comments

**Comment by ChronosSF** [comment]
@HUSSAR-mtrela , this seems to work nice now. I am not adding more tests as the idea was mostly to have a testing library that can read the md files and execute on the results. If you can think of any tests other than checking for replacement words , please let me know so we can put them in the backlog for development.

**Review by HUSSAR-mtrela — APPROVED** [review]

**Pull request merged**